src/components/AuthContainer.js

```diff
@@ -22,15 +22,21 @@ function getUserIsRegisteredAndEnabled() {
 }
 
 async function createClusters() {
+  const userProfile = Utils.getUser().getBasicProfile()
   const [billingProjects, clusters] = await Promise.all(
     [Rawls.listBillingProjects(), Leo.clustersList()])
   const projectsWithoutClusters = _.difference(
-    _.map(billingProjects, 'projectName'),
-    _.map(clusters, 'googleProject')
+    _.uniq(_.map(billingProjects, 'projectName')), // in case of being both a user and an admin of a project
+    _.map(
+      _.filter(clusters, c => c.creator === userProfile.getEmail()),
+      'googleProject'
+    )
   )
 
   projectsWithoutClusters.forEach(project => {
-    Leo.cluster(project, `launchpad-${project}`).create({
+    Leo.cluster(project,
+      `saturn-${userProfile.getId()}-${project}`.match(/(?:[a-z](?:[-a-z0-9]{0,49}[a-z0-9])?)/)[0] // regex used by google for valid names
```

Comment:
Why are we sending this string through a regex match?

Reply:
Google will complain if the name is too long, or if it ends with a non-alphanumerical character (i.e. a dash). This is the regex the failure return says the name must match again. To make sure it does, we just take the part that fits.